pkg/cloud/aws/services/ec2/vpc.go

```diff
@@ -140,25 +158,17 @@ func (s *Service) createVPC() (*v1alpha1.VPCSpec, error) {
 	}
 	s.scope.V(2).Info("Created new VPC with cidr", "vpc-id", *out.Vpc.VpcId, "cidr-block", *out.Vpc.CidrBlock)
 
+	// Record VPC ID as soon as we possibly can, so that we persist the ID in the case of later failure
+	s.scope.VPC().ID = aws.StringValue(out.Vpc.VpcId)
```

Comment:
@sethp-nr I believe this addresses your concern about not recording the VPC ID in the case of later failure.

Reply:
After further testing, if we persist the vpc ID without tagging, subsequent reconciles will treat the VPC as an "unmanaged VPC", I'm going to add a TODO and notes, but will remove this for now, since this has the potential to create hopelessly broken clusters, which may just be worse than orphaning a VPC.

Reply:
@vincepri as we refactor for v1alpha2, we likely want to keep this issue in mind. We should have a way to differentiate between managed and unmanaged VPCs based on more than just tagging.

Reply:
Could we add a field that indicates it was created by CAPA and set it here too?

Reply:
@ncdc I was thinking we could potentially update the current Cluster.Spec.ProviderSpec.NetworkSpec to be either a ManagedNetworkSpec or a UnmanagedNetworkSpec (maybe ExternalNetworkSpec), This would also potentially simplify config for users providing an UnmanagedNetworkSpec or ExternalNetworkSpec to avoid exposing them to fields that are not applicable to them.

Reply:
That said, it would definitely complicate some of the downstream logic for consuming Managed vs Unmanaged NetworkSpecs...

Reply:
We could have a helper function that gets a generic network spec from the unioned bit (e.g. pick either ...networkSpec.Managed or ...networkSpec.Unmanaged) and use that helper everywhere?